Sources/System.swift

```diff
@@ -27,17 +27,15 @@ class System {
     static var name: String? {
         var systemInfo = utsname()
         uname(&systemInfo)
-
-        if let identifier = String(cString: &systemInfo.machine.0, encoding: .ascii) {
-            // Simulator Check
+        if let identifier = withUnsafePointer(to: &systemInfo.machine, {
+            $0.withMemoryRebound(to: CChar.self, capacity: 1) {
+                ptr in String.init(validatingUTF8: ptr)
+            }}) {
```

Comment:
I would suggest alternative (cleaner) solution:
```suggestion
        if let identifier = String(cString: UnsafePointer(&systemInfo.machine.0), encoding: .ascii) {
```

Reply:
This will cause a warning:
>Initialization of 'UnsafePointer<CChar>' (aka 'UnsafePointer<Int8>') results in a dangling pointer